Doc: Add at-id reference tag to Strided Arrays interface section

mbauman opened the pull request.

mbauman committed:
Doc: Add at-id reference tag to Strided Arrays interface section

[ci skip]

fredrikekre commented on doc/src/manual/interfaces.md:
```diff
@@ -391,7 +391,7 @@ something other than 1), you should specialize `indices`. You should also specia
 perhaps range-types `Ind` of your own design. For more information, see [Arrays with custom indices](@ref).
 
-## [Strided Arrays]
+## [Strided Arrays](@id man-interface-strided-arrays)
```
Is this tag used somewhere or what is the reason for this?

mbauman replied:
No, I'm just following the convention in this file — and more importantly getting rid of the literal `[]` in the header.

fredrikekre replied:
> ... getting rid of the literal [] in the header

Ah, right! So the alternative fix would just be to remove those I guess :)

mbauman replied:
Or actually link to it. :)

mbauman committed:
Link to array and strided array interface sections

This "array implementation" section is old and should probably eventually be folded into the interfaces chapter, but I think this incremental improvement here is worth doing in lieu of a larger overhaul.
